csai/src/main/java/com/dice/doris/csai/MainActivity.kt

```diff
@@ -39,44 +40,44 @@ class MainActivity : AppCompatActivity(), SourceCallback {
         playerView.setShowPreviousButton(false)
         playerView.setShowNextButton(false)
         playerView.setShowSubtitleButton(true)
+        secondaryPlayerView.keepScreenOn = true
         SourceUtils.getSource(this, this)
     }
 
     override fun onSourceCallback(videoInfo: VideoInfo, adsConfiguration: AdsConfiguration?) {
         val imaCsaiProperties = parseCsaiProperties(adsConfiguration)
 
-        val src = SourceBuilder()
+        val source = SourceBuilder()
             .setId(CsaiConfig.videoId)
             .setUrl(videoInfo.url)
             .setDrmParams(videoInfo.drm?.let { drm -> ActionToken(CsaiConfig.videoId, videoInfo.url, drm.url, drm.jwtToken, "widevine") })
             .setImaCsaiProperties(imaCsaiProperties)
             .setTextTracks(parseTextTrack(videoInfo))
             .build()
 
-        player = createPlayer(Source.getAdType(src))
+        val adType = Source.getAdType(source)
+        // Currently we use two player views for csai live playback because can not get SurfaceView.
+        // When our doris ui supports TV device and then we should not use two views.
+        val adViewProvider: AdViewProvider =
+            if (adType == AdType.IMA_CSAI_LIVE) secondaryPlayerView else playerView
+        player = createPlayer(adType, adViewProvider)
+        player?.setDorisListener(this)
 
         player?.addAnalyticsListener(EventLogger())
 
-        player?.addListener(object : Player.Listener {
-            override fun onRenderedFirstFrame() {
-                progressBar.visibility = View.GONE
-            }
-        })
-
         playerView.player = player?.exoPlayer
 
-        player?.load(src)
+        player?.load(source)
     }
 
-    private fun createPlayer(adType: DorisAdEvent.AdType): ExoDoris {
-        val builder = if (adType === DorisAdEvent.AdType.IMA_CSAI) {
+    private fun createPlayer(adType: AdType, adViewProvider: AdViewProvider): ExoDoris {
+        val builder = if (adType === AdType.IMA_CSAI) {
             ExoDorisImaCsaiBuilder(this@MainActivity).apply {
-                setAdViewProvider(playerView)
+                setAdViewProvider(adViewProvider)
             }
-        } else if (adType === DorisAdEvent.AdType.IMA_CSAI_LIVE) {
+        } else if (adType === AdType.IMA_CSAI_LIVE) {
             ExoDorisImaCsaiLiveBuilder(this@MainActivity).apply {
-                setLiveAdSurfaceView(playerView.videoSurfaceView as SurfaceView)
-                setAdViewProvider(playerView)
+                setAdViewProvider(adViewProvider)
```

Comment:
seems `adViewProvider` is not needed.

```suggestion
                setAdViewProvider(secondaryPlayerView)
```